gen.go

```diff
@@ -19,10 +20,38 @@ const codeTemplate = `
 
 package multicodec
 
-const ({{ range . }}
-// {{ if .IsDeprecated }}Deprecated: {{ end }}{{ .VarName }} is a {{ .Status }} code tagged "{{ .Tag }}"{{ if .Description }} and described by: {{ .Description }}{{ end }}.
-{{ .VarName }} Code = {{ .Code }} // {{ .Name }}
-{{ end }})
+const (
+{{- range .Entries }}
+	// {{ if .IsDeprecated }}Deprecated: {{ end }}{{ .ConstName }} is a {{ .Status }} code tagged "{{ .Tag }}"{{ if .Description }} and described by: {{ .Description }}{{ end }}.
+	{{ .ConstName }} Code = {{ .Code }} // {{ .Name }}
+{{ end -}}
+)
+
+var knownCodes = []Code{
+{{- range .Entries }}
+	{{ .ConstName }},
+{{- end }}
+}
+
+func (c Code) Tag() string {
```

Comment:
Can these tags be declared as constants (similar to the codes) to quickly discover the available/existing ones? (instead of needing to browse the long switch)

Reply:
(just a nit, not blocking on my end)

Reply:
I'm not sure; in the original issue I pondered whether this should be an enum rather than a string, but I also don't think upstream treats tags as fixed, stable constants. They're just metadata strings, as far as I can tell. I worry that using enums would give the false sense that they're indeed fixed forever, just like the multicodec integers are.

Maybe @rvagg @vmx or @willscott have thoughts.

If we don't want to declare an enum, we could still make the enumeration of known tags public, via something like `func KnownTags() string` :) I initially thought that would be unnecessary, but I'm happy to add it if go-ipfs needs it. What use case do you have? I thought you'd essentially just query tags and compare them against known ones like `ipld`.

Reply:
Yes, metadata rather than "fixed, stable constants", there's some amount of ambivalence toward the tags because categorisation is a tricky thing in many cases so the edges are squishy (as demonstrated by this serialization->ipld switch). I'd prefer that they not be raised too much in prominence in our APIs.

We're going to switch from a CSV to a more flexible format for input at some point (CSV will still exist, but a JSON can hold much more information), so that will give us additional ability to describe and categorise entries. Perhaps even having multiple tags for individual entries. So for now, don't push "tag" too far.

Reply:
Again, this is just a nit, feel free to ignore. The suggested pattern is not having hard-coded strings scattered in a giant switch, but this is generated code from a spec'ed table so it's fine.

Reply:
I'm going to keep it as a string for now, then. If the upstream CSV becomes richer, or starts declaring tags as fixed constants, then I'll be more than happy to update the API to reflect that.